Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCRtpTransformBackend.cpp

```diff
@@ -42,20 +42,34 @@ void LibWebRTCRtpTransformBackend::clearTransformableFrameCallback()
     setInputCallback({ });
 }
 
-void LibWebRTCRtpTransformBackend::setOutputCallback(rtc::scoped_refptr<webrtc::TransformedFrameCallback>&& callback)
+void LibWebRTCRtpTransformBackend::addOutputCallback(rtc::scoped_refptr<webrtc::TransformedFrameCallback>&& callback, uint32_t ssrc)
 {
-    Locker locker { m_outputCallbackLock };
-    m_outputCallback = WTFMove(callback);
+    Locker locker { m_outputCallbacksLock };
+    m_outputCallbacks.insert_or_assign(ssrc, WTFMove(callback));
 }
 
-void LibWebRTCRtpTransformBackend::processTransformedFrame(RTCRtpTransformableFrame& frame)
+void LibWebRTCRtpTransformBackend::removeOutputCallback(uint32_t ssrc)
 {
-    auto rtcFrame = static_cast<LibWebRTCRtpTransformableFrame&>(frame).takeRTCFrame();
-    if (!rtcFrame)
+    Locker locker { m_outputCallbacksLock };
+    m_outputCallbacks.erase(ssrc);
+}
+
+void LibWebRTCRtpTransformBackend::sendFrameToOutput(std::unique_ptr<webrtc::TransformableFrameInterface>&& rtcFrame)
+{
+    Locker locker { m_outputCallbacksLock };
+    if (m_outputCallbacks.size() == 1) {
+        m_outputCallbacks.begin()->second->OnTransformedFrame(WTFMove(rtcFrame));
```

Comment:
would it worth releasing the lock before calling OnTransformedFrame so that we don't block whatever pending call this could be ?

Reply:
I thought about this and decided to keep what we are doing currently on the basis that there is no real deadlock risk and this would add some count churn. Either way is fine.